it/v2/endpoints/DeleteHistoricNonFhlUkPropertyAnnualSubmissionControllerISpec.scala

```diff
@@ -0,0 +1,139 @@
+          MtdIdLookupStub.ninoFound(nino)
+          DownstreamStub.onSuccess(DownstreamStub.PUT, desUri, Status.NO_CONTENT, JsObject.empty)
+        }
+
+        val response: WSResponse = await(request().delete())
+        response.status shouldBe Status.NO_CONTENT
+        response.header("X-CorrelationId").nonEmpty shouldBe true
+      }
+    }
+
+    "return error according to spec" when {
+
+      "a validation error occurs" when {
+        def validationErrorTest(requestNino: String, requestTaxYear: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"validation fails with ${expectedBody.code} error" in new Test {
+
+            override val nino: String    = requestNino
+            override val taxYear: String = requestTaxYear
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+            }
+
+            val response: WSResponse = await(request().delete())
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          ("hello", "2021-22", Status.BAD_REQUEST, NinoFormatError),
+          ("AA123456A", "Beans", Status.BAD_REQUEST, TaxYearFormatError),
+          ("AA123456A", "2021-23", Status.BAD_REQUEST, RuleTaxYearRangeInvalidError),
+          ("AA123456A", "2016-17", Status.BAD_REQUEST, RuleHistoricTaxYearNotSupportedError)
+        )
+        input.foreach(args => (validationErrorTest _).tupled(args))
+      }
+
+      "des service error" when {
+        def serviceErrorTest(desStatus: Int, desCode: String, expectedStatus: Int, expectedBody: MtdError): Unit = {
+          s"des returns an $desCode error and status $desStatus" in new Test {
+
+            override def setupStubs(): StubMapping = {
+              AuditStub.audit()
+              AuthStub.authorised()
+              MtdIdLookupStub.ninoFound(nino)
+              DownstreamStub.onError(DownstreamStub.PUT, desUri, desStatus, errorBody(desCode))
+            }
+
+            val response: WSResponse = await(request().delete())
+            response.status shouldBe expectedStatus
+            response.json shouldBe Json.toJson(expectedBody)
+          }
+        }
+
+        val input = Seq(
+          (Status.BAD_REQUEST, "INVALID_NINO", Status.BAD_REQUEST, NinoFormatError),
+          (Status.BAD_REQUEST, "INVALID_TAX_YEAR", Status.BAD_REQUEST, TaxYearFormatError),
+          (Status.BAD_REQUEST, "INVALID_CORRELATIONID", Status.INTERNAL_SERVER_ERROR, InternalError),
+          (Status.NOT_FOUND, "NOT_FOUND", Status.NOT_FOUND, NotFoundError),
+          (Status.NOT_FOUND, "NOT_FOUND_PROPERTY", Status.NOT_FOUND, NotFoundError),
+          (Status.GONE, "GONE", Status.NOT_FOUND, NotFoundError),
+          (Status.UNPROCESSABLE_ENTITY, "UNPROCESSABLE_ENTTY", Status.INTERNAL_SERVER_ERROR, InternalError),
```

Comment:
There is no unprocessable_entity in the tech spec? Perhaps remove line 131?

Reply:
Probably needs updating in the FHL Controller ISpec too, since they use the same backend spec

Reply:
We may need to add in a test for "INVALID_PAYLOAD" here too

Reply:
Line 131 - error status can be Status.UNPROCESSABLE_ENTITY, but name should be "TAX_YEAR_NOT_SUPPORTED" ?

Reply:
> We may need to add in a test for "INVALID_PAYLOAD" here too

- No longer necessary now that we have clarified that the backend is IFS and not DES